pkg/capture/capturetypes/packet.go

```diff
@@ -115,22 +112,24 @@ func classifyTCP(packet *GPPacket) Direction {
 		}
 	}
 
-	return classifyByPorts(packet)
+	return classifyByPorts(epHash)
 }
 
-func classifyUDP(packet *GPPacket) Direction {
-	// handle broadcast / multicast addresses (we do not need to check the
+func classifyUDP(epHash EPHash, isIPv4 bool) Direction {
+
+	// Handle broadcast / multicast addresses (we do not need to check the
 	// inverse direction because it won't be in multicast format)
-	if isBroadcastMulticast(packet.EPHash[16:32], packet.IsIPv4) {
+	if isBroadcastMulticast(epHash[16:32], isIPv4) {
 		return DirectionRemains
 	}
 
-	return classifyByPorts(packet)
+	return classifyByPorts(epHash)
 }
 
-func classifyICMPv4(packet *GPPacket) Direction {
+func classifyICMPv4(icmpType byte) Direction {
+
 	// Check the ICMPv4 Type parameter
-	switch packet.AuxInfo {
+	switch icmpType {
 
 	// EchoReply, DestinationUnreachable, TimeExceeded, ParameterProblem, TimestampReply
 	case 0x00, 0x03, 0x0B, 0x0C, 0x0E:
```

Comment:
Just saw this. Should be `const`s to increase readability.

Reply:
Not sure it really increases readability (because the numbers are different for IPv4 and IPv6 so we're basically only adding a lot of lines of code for an internal-only piece of code) but I don't have strong feelings about that. Will adapt.